File: tests/0226_invert_binary_tree_test.py
import importlib
import pytest
from algorithms.tree_node import make_tree

module = importlib.import_module("algorithms.0226_invert_binary_tree")


def test_invert_tree():
    t = make_tree([4, 2, 7, 1, 3, 6, 9])

    s = module.Solution()
    r = s.invertTree(t[0])

    assert r.val == 4
    assert r.left.val == 7
    assert r.right.val == 2
    assert r.left.left.val == 9
    assert r.left.right.val == 6
    assert r.right.left.val == 3
    assert r.right.right.val == 1

File: tests/0226_invert_binary_tree_test.cpp
#include "0226_invert_binary_tree.cpp"
#include <doctest.h>

TEST_CASE("invertTree")
{
    auto t = tree::make_tree({4, 2, 7, 1, 3, 6, 9});

    Solution s;
    auto r = s.invertTree(t[0].get());

    REQUIRE(r != nullptr);
    REQUIRE(r->left != nullptr);
    REQUIRE(r->right != nullptr);
    REQUIRE(r->left->val == 7);
    REQUIRE(r->right->val == 2);
    REQUIRE(r->left->left != nullptr);
    REQUIRE(r->left->right != nullptr);
    REQUIRE(r->left->left->val == 9);
    REQUIRE(r->left->right->val == 6);
    REQUIRE(r->right->left != nullptr);
    REQUIRE(r->right->right != nullptr);
    REQUIRE(r->right->left->val == 3);
    REQUIRE(r->right->right->val == 1);
}
